### ISSUE TITLE
job local cache not getting updated on 2015.2 

### ISSUE THREAD

**Issue by mimianddaniel** [saltstack/salt#21702]
I am not able to look up local job using `salt-run jobs.lookup_jid` now... because of this.

```python
2015-03-11 15:45:37,816 [salt.master                              ][ERROR   ][17713] Error in function _syndic_return:
Traceback (most recent call last):
  File "/opt/blue-python/2.7/lib/python2.7/site-packages/salt/master.py", line 1415, in run_func
    ret = getattr(self, func)(load)
  File "/opt/blue-python/2.7/lib/python2.7/site-packages/salt/master.py", line 1270, in _syndic_return
    self._return(ret)
  File "/opt/blue-python/2.7/lib/python2.7/site-packages/salt/master.py", line 1244, in _return
    self.opts, load, event=self.event, mminion=self.mminion)
  File "/opt/blue-python/2.7/lib/python2.7/site-packages/salt/utils/job.py", line 53, in store_job
    load.update({'fun': ret_['fun']})
TypeError: string indices must be integers, not str
```
```yaml
          Salt: 2015.2.0
         Python: 2.7.9 (default, Dec 30 2014, 13:07:54)
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.4.1
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
```

**Comment by jfindlay** [comment]
@mimianddaniel, I am seeing the same error, but haven't been using `salt-run jobs.lookup_jid`.

**Comment by mimianddaniel** [comment]
Hi,
we are not using separate returner at the moment but have salteventsd we
can look up if required but wouldnt this be considered core feature of salt
and should be fixed before the general release?

**Comment by basepi** [comment]
I think the syndic fixes in 2014.7 that were merged forward have fixed this issue. Neither @jfindlay nor I can reproduce on the current head of 2015.2 (g8a301bc). I'm going to close this for now -- if you test and fine differently, please comment and we'll re-open!
